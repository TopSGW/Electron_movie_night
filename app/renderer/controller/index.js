// Logic for mapping application state onto component props.
import {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  SearchMovies,
  MainContent,
  MovieThumbnailContainer
} from './containers'

// Logic for mapping application events directly to redux dispatch actions.
import './mapEventsToDispatch'

// Re-export all the containers.
export {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  MainContent,
  MovieThumbnailContainer,
  SearchMovies
}
