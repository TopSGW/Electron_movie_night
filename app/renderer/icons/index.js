import cn from 'classnames'
import React from 'react'

const Icon = ({ className, border, fixedWidth, large, pulse, size3x, ...props }) => {
  return <i
    className={cn(
      'fa',
      className,
      pulse && 'fa-pulse',
      large && 'fa-lg',
      fixedWidth && 'fa-fw',
      size3x && 'fa-3x',
      border && 'fa-border'
    )}
    {...props}
  />
}

export const Anchor = ({ className, ...props }) => {
  return <Icon className={cn('fa-anchor', className)} {...props} />
}

export const Angle = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-angle-down',
      left && 'fa-angle-left',
      right && 'fa-angle-right',
      up && 'fa-angle-up',
      !down && !left && !right && !up && 'fa-angle-right', // default
      className
    )}
    {...props}
  />
}

export const Bars = ({ className, ...props }) => {
  return <Icon className={cn('fa-bars', className)} {...props} />
}

export const Ban = ({ className, ...props }) => {
  return <Icon className={cn('fa-ban', className)} {...props} />
}

export const Chevron = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-chevron-down',
      left && 'fa-chevron-left',
      right && 'fa-chevron-right',
      up && 'fa-chevron-up',
      !down && !left && !right && !up && 'fa-chevron-right', // default
      className
    )}
    {...props}
  />
}

export const Database = ({ className, ...props }) => {
  return <Icon className={cn('fa-database', className)} {...props} />
}

export const Close = ({ className, ...props }) => {
  return <Icon className={cn('fa-close', className)} {...props} />
}

export const Edit = ({ className, ...props }) => {
  return <Icon className={cn('fa-pencil', className)} {...props} />
}

export const FileIcon = ({ className, video, ...props }) => {
  return <Icon
    className={cn(
      video && 'fa-file-video-o',
      !video && 'fa-file', // default
      className
    )}
    {...props}
  />
}

export const ImportIcon = ({ className, ...props }) => {
  return <Icon
    className={cn(
      'fa-download',
      className
    )}
    {...props}
  />
}

export const Info = ({ className, circle, ...props }) => {
  return <Icon
    className={cn(
      circle && 'fa-info-circle',
      !circle && 'fa-info',
      className
    )}
    {...props}
  />
}

export const Play = ({ className, ...props }) => {
  return <Icon className={cn('fa-play', className)} {...props} />
}

export const Search = ({ className, ...props }) => {
  return <Icon className={cn('fa-search', className)} {...props} />
}

export const Spinner = ({ className, ...props }) => {
  return <Icon className={cn('fa-spinner', className)} {...props} />
}

export const Star = ({ className, half, outline, ...props }) => {
  return <Icon
    className={cn(
      half && 'fa-star-half-o',
      outline && 'fa-star-o',
      !half && !outline && 'fa-star', // default
      className
    )}
    {...props}
  />
}

export const Trash = ({ className, ...props }) => {
  return <Icon className={cn('fa-trash', className)} {...props} />
}
