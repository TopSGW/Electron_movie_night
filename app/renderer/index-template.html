<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <title>Movie Night</title>
    <link rel="stylesheet" href="../vendor/font-awesome-4.7.0/css/font-awesome.min.css">
    <%=htmlWebpackPlugin.options.css%>
  </head>
  <body style="background-color:rgb(20,20,20) !important; /* override bootstrap */">
    <div id="root"></div>
    <script src="<%=htmlWebpackPlugin.options.bundle%>"></script>
  </body>
</html>
