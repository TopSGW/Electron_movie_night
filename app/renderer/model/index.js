import { ipcRenderer } from 'electron'
import { createStore, applyMiddleware, combineReducers } from 'redux'
import reduxLogger from 'redux-logger'

import { MOVE_MOVIE_TO_TRASH, UPDATE_MOVIE_METADATA } from '../../shared/events'
import { isDevEnv } from '../../shared/utils'
import appLogger from '../mainWindowLogger'

// Reducers
import dbLoaded from './dbLoaded'
import crawl from './crawl'
import featuredMovie from './featuredMovie'
import movies from './movies'
import search from './search'

// Top level state selectors.
import {
  crawlActiveSelector,
  crawlStatsSelector,
  showCrawlStatsSelector,
  movieDBSelector,
  searchCategorySelector,
  searchQuerySelector
} from './selectors'

// Memoized Selectors
import enhancedFeaturedMovieSelector from './enhancedFeaturedMovie'
import visibleMoviesSelector from './visibleMovies'

// Re-export all the action creators.
// ----------------------------------
export * from './actions'

// Electron action creators.
// -------------------------
export const updateMetadataFor = (movie) => {
  ipcRenderer.send(UPDATE_MOVIE_METADATA, movie)
  appLogger.info('Sent UPDATE_MOVIE_METADATA event', { movie: movie.title })
}

export const deleteMovie = (movie) => {
  ipcRenderer.send(MOVE_MOVIE_TO_TRASH, movie)
  appLogger.info(`Sent ${MOVE_MOVIE_TO_TRASH} event`, { movie: movie.title })
}

// Public selectors.
// These allow connected containers to depend on state
// without requiring knowledge of the state's shape.
// ----------------------------------------------------------------------
export const getAllMovies = (state) => {
  return movieDBSelector(state)
}

export const getCrawlActive = (state) => {
  return crawlActiveSelector(state)
}

export const getCrawlStats = (state) => {
  return crawlStatsSelector(state)
}

export const getShowCrawlStats = (state) => {
  return showCrawlStatsSelector(state)
}

export const getFeaturedMovie = (state) => {
  return enhancedFeaturedMovieSelector(state)
}

export const getSearchCategory = (state) => {
  return searchCategorySelector(state)
}

export const getSearchQuery = (state) => {
  return searchQuerySelector(state)
}

export const getVisibleMovies = (state) => {
  return visibleMoviesSelector(state)
}

// Export function for creating redux store.
// -----------------------------------------
export const createReduxStore = () => {
  // Create the redux store.
  // This is the default export.
  // ---------------------------
  let middlewares = []
  if (isDevEnv()) {
    middlewares.push(reduxLogger)
  }

  const rootReducer = combineReducers({
    crawl,
    dbLoaded,
    featuredMovie,
    movies,
    search
  })

  const store = createStore(rootReducer, applyMiddleware(...middlewares))
  return store
}

// This is the actual store used in the app.
// -----------------------------------------
const store = createReduxStore()
export default store
