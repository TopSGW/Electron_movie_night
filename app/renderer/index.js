import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { logEnv } from '../shared/utils'
import { App } from './controller'
import store from './model'
import logger from './mainWindowLogger'

logEnv(logger)

// Render the App.
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
