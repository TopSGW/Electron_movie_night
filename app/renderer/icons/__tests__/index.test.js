'use strict'
/* globals describe, test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import * as Icons from '../index'

describe('font awesome icons', () => {
  test('classname passed through', () => {
    const cn = 'orig-class-name'
    const component = renderer.create(
      <Icons.Anchor className={cn} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.type).toBe('i')
    expect(tree.props.className).toBe(`fa fa-anchor ${cn}`)
  })

  test('enhancers', () => {
    const enhancers = {
      border: 'fa-border',
      fixedWidth: 'fa-fw',
      large: 'fa-lg',
      pulse: 'fa-pulse',
      size3x: 'fa-3x'
    }

    Object.keys(enhancers).forEach(enhancer => {
      const dynamicProp = { [enhancer]: true }
      const component = renderer.create(
        <Icons.Anchor {...dynamicProp} />
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree.type).toBe('i')
      expect(tree.props.className).toBe(`fa fa-anchor ${enhancers[enhancer]}`)
    })
  })

  test('icons', () => {
    Object.keys(Icons).forEach(key => {
      let Tag = Icons[key]
      const component = renderer.create(
        <Tag />
      )

      let expected = key.toLowerCase()
      if (expected === 'fileicon') { expected = 'fa-file' }
      if (expected === 'edit') { expected = 'fa-pencil' }
      if (expected === 'importicon') { expected = 'fa-download' }

      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree.type).toBe('i')
      expect(tree.props.className).toContain(expected)
    })
  })

  test('Angle enhancers', () => {
    const directions = ['down', 'left', 'right', 'up']
    directions.forEach(direction => {
      const dynamicProp = { [direction]: true }
      const component = renderer.create(
        <Icons.Angle {...dynamicProp} />
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree.type).toBe('i')
      expect(tree.props.className).toBe(`fa fa-angle-${direction}`)
    })
  })

  test('Chevron enhancers', () => {
    const directions = ['down', 'left', 'right', 'up']
    directions.forEach(direction => {
      const dynamicProp = { [direction]: true }
      const component = renderer.create(
        <Icons.Chevron {...dynamicProp} />
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree.type).toBe('i')
      expect(tree.props.className).toBe(`fa fa-chevron-${direction}`)
    })
  })

  test('FileIcon video', () => {
    const component = renderer.create(
      <Icons.FileIcon video />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.type).toBe('i')
    expect(tree.props.className).toBe(`fa fa-file-video-o`)
  })

  test('Info circle', () => {
    const component = renderer.create(
      <Icons.Info circle />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.type).toBe('i')
    expect(tree.props.className).toBe(`fa fa-info-circle`)
  })

  test('Star half', () => {
    const component = renderer.create(
      <Icons.Star half />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.type).toBe('i')
    expect(tree.props.className).toBe(`fa fa-star-half-o`)
  })

  test('Star outline', () => {
    const component = renderer.create(
      <Icons.Star outline />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.type).toBe('i')
    expect(tree.props.className).toBe(`fa fa-star-o`)
  })
})
