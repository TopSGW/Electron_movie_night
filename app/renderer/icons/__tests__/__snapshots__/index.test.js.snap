// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`font awesome icons Angle enhancers 1`] = `
<i
  className="fa fa-angle-down"
/>
`;

exports[`font awesome icons Angle enhancers 2`] = `
<i
  className="fa fa-angle-left"
/>
`;

exports[`font awesome icons Angle enhancers 3`] = `
<i
  className="fa fa-angle-right"
/>
`;

exports[`font awesome icons Angle enhancers 4`] = `
<i
  className="fa fa-angle-up"
/>
`;

exports[`font awesome icons Chevron enhancers 1`] = `
<i
  className="fa fa-chevron-down"
/>
`;

exports[`font awesome icons Chevron enhancers 2`] = `
<i
  className="fa fa-chevron-left"
/>
`;

exports[`font awesome icons Chevron enhancers 3`] = `
<i
  className="fa fa-chevron-right"
/>
`;

exports[`font awesome icons Chevron enhancers 4`] = `
<i
  className="fa fa-chevron-up"
/>
`;

exports[`font awesome icons FileIcon video 1`] = `
<i
  className="fa fa-file-video-o"
/>
`;

exports[`font awesome icons Info circle 1`] = `
<i
  className="fa fa-info-circle"
/>
`;

exports[`font awesome icons Star half 1`] = `
<i
  className="fa fa-star-half-o"
/>
`;

exports[`font awesome icons Star outline 1`] = `
<i
  className="fa fa-star-o"
/>
`;

exports[`font awesome icons classname passed through 1`] = `
<i
  className="fa fa-anchor orig-class-name"
/>
`;

exports[`font awesome icons enhancers 1`] = `
<i
  className="fa fa-anchor fa-border"
/>
`;

exports[`font awesome icons enhancers 2`] = `
<i
  className="fa fa-anchor fa-fw"
/>
`;

exports[`font awesome icons enhancers 3`] = `
<i
  className="fa fa-anchor fa-lg"
/>
`;

exports[`font awesome icons enhancers 4`] = `
<i
  className="fa fa-anchor fa-pulse"
/>
`;

exports[`font awesome icons enhancers 5`] = `
<i
  className="fa fa-anchor fa-3x"
/>
`;

exports[`font awesome icons icons 1`] = `
<i
  className="fa fa-anchor"
/>
`;

exports[`font awesome icons icons 2`] = `
<i
  className="fa fa-angle-right"
/>
`;

exports[`font awesome icons icons 3`] = `
<i
  className="fa fa-bars"
/>
`;

exports[`font awesome icons icons 4`] = `
<i
  className="fa fa-ban"
/>
`;

exports[`font awesome icons icons 5`] = `
<i
  className="fa fa-chevron-right"
/>
`;

exports[`font awesome icons icons 6`] = `
<i
  className="fa fa-database"
/>
`;

exports[`font awesome icons icons 7`] = `
<i
  className="fa fa-close"
/>
`;

exports[`font awesome icons icons 8`] = `
<i
  className="fa fa-pencil"
/>
`;

exports[`font awesome icons icons 9`] = `
<i
  className="fa fa-file"
/>
`;

exports[`font awesome icons icons 10`] = `
<i
  className="fa fa-download"
/>
`;

exports[`font awesome icons icons 11`] = `
<i
  className="fa fa-info"
/>
`;

exports[`font awesome icons icons 12`] = `
<i
  className="fa fa-play"
/>
`;

exports[`font awesome icons icons 13`] = `
<i
  className="fa fa-search"
/>
`;

exports[`font awesome icons icons 14`] = `
<i
  className="fa fa-spinner"
/>
`;

exports[`font awesome icons icons 15`] = `
<i
  className="fa fa-star"
/>
`;

exports[`font awesome icons icons 16`] = `
<i
  className="fa fa-trash"
/>
`;
